### PR TITLE
Refactor- change user to person

### PR DESCRIPTION
**Pull request by DavidHeymann**
- Change user model to person
- Mongoose decided to call the collection "people", because in his opinion, "persons" is the wrong word, although according to https://dictionary.cambridge.org/grammar/british-grammar/person-persons-or-people this is a more correct word

### CODE REVIEW

File: src/person/person.interface.ts [hunk 1]
```diff
@@ -3,13 +3,13 @@ import { Rank, Responsibility } from '../utils';
...
-export interface IUser extends Document {
-// User's Basic information
+export interface IPerson extends Document {
+// Person's Basic information
   _id: ObjectId;
   identityCard: string;
   personalNumber: string;
-  primaryUser: string;
-  secondaryUsers: string[];
+  primaryPerson: string;
```
Comment (eladex) [inline src/person/person.interface.ts]: Should be "primaryUser" or "primaryDomainUser"

File: src/person/person.interface.ts [hunk 2]
```diff
@@ -3,13 +3,13 @@ import { Rank, Responsibility } from '../utils';
...
+  secondaryPersons: string[];
```
Comment (eladex) [inline src/person/person.interface.ts]: Should be "domainUsers" or "secondaryDomainUsers"

File: src/person/person.interface.ts [hunk 3]
```diff
@@ -35,6 +35,6 @@ export interface IUser extends Document {
 }
 
 export const PERSONAL_FIELDS = ['job', 'mail', 'phone', 'address', 'mobilePhone'];
-export const USER_FIELDS = PERSONAL_FIELDS.concat(
-  ['primaryUser', 'secondaryUsers', 'serviceType', 'firstName', 'lastName', 'currentUnit', 
+export const PERSON_FIELDS = PERSONAL_FIELDS.concat(
```
Comment (eladex) [inline src/person/person.interface.ts]: should think of another name

File: src/person/person.model.ts [hunk 4]
```diff
@@ -1,51 +1,51 @@
...
     },
     personalNumber: {
       type: String,
       unique: true,
       sparse: true,
-      validate: { validator: UserValidate.personalNumber, message: '{VALUE} is an invalid personal number!' },
+      validate: { validator: PersonValidate.personalNumber, message: '{VALUE} is an invalid personal number!' },
     },
-    primaryUser: {
+    primaryPerson: {
```
Comment (eladex) [inline src/person/person.model.ts]: Should be "primaryUser" or "primaryDomainUser"

File: src/person/person.model.ts [hunk 5]
```diff
@@ -1,51 +1,51 @@
...
       type: String,
-      required: [true, 'You must enter a primary user!'],
+      required: [true, 'You must enter a primary person!'],
       unique: true,
-      validate: { validator: UserValidate.email, message: '{VALUE} is an invalid User' },
+      validate: { validator: PersonValidate.email, message: '{VALUE} is an invalid Person' },
     },
-    secondaryUsers: [{
+    secondaryPersons: [{
```
Comment (eladex) [inline src/person/person.model.ts]: Should be "domainUsers" or "secondaryDomainUsers"

File: src/person/person.route.spec.ts [hunk 6]
```diff
@@ -4,54 +4,54 @@ process.env.PORT = '8080';
...
 const expect = chai.expect;
 
-const userExamples: IUser[] = [  
-  <IUser>{
+const personExamples: IPerson[] = [  
+  <IPerson>{
     identityCard: '234567891',
     personalNumber: '3456712',
-    primaryUser: 'mazaltov@surprise.sod',
+    primaryPerson: 'mazaltov@surprise.sod',
```
Comment (eladex) [inline src/person/person.route.spec.ts]: Same here, the name should be changed across all tests

File: src/person/person.spec.ts [hunk 7]
```diff
@@ -0,0 +1,739 @@
...
+const expect = chai.expect;
+chai.use(require('chai-http'));
+
+const dbIdExample = ['5b50a76713ddf90af494de32', '5b56e5ca07f0de0f38110b9c'];
+
+const personExamples: IPerson[] = [
+  <IPerson>{
+    identityCard: '123456789',
+    personalNumber: '2345671',
+    primaryPerson: 'aviron@secure.sod',
```
Comment (eladex) [inline src/person/person.spec.ts]: also in this file...

File: src/person/person.interface.ts (hunk 2 above)
Comment (DavidHeymann) [reply src/person/person.interface.ts]: Done

Comment (DavidHeymann) [reply src/person/person.interface.ts]: Done

File: src/person/person.model.ts (hunk 4 above)
Comment (DavidHeymann) [reply src/person/person.model.ts]: Done

File: src/person/person.model.ts (hunk 5 above)
Comment (DavidHeymann) [reply src/person/person.model.ts]: Done

File: src/person/person.route.spec.ts (hunk 6 above)
Comment (DavidHeymann) [reply src/person/person.route.spec.ts]: Done

File: src/person/person.spec.ts (hunk 7 above)
Comment (DavidHeymann) [reply src/person/person.spec.ts]: Done

File: src/person/person.interface.ts (hunk 3 above)
Comment (DavidHeymann) [reply src/person/person.interface.ts]: Done

**Pull request merged**